### PR TITLE
BugFix: Network-25372 : Global Secure Access client is deployed on all managed endpoints

### LINKED ISSUE #960: Netspec 25372 - Device Count calculation seems broken
When no Intune managed devices are detected, we we failed to calculate the % value and mark the check as investigate, instead of failed:

Image

### PR DESCRIPTION
**Pull request by ashwinikarke**

### CODE REVIEW

File: src/powershell/tests/Test-Assessment.25372.ps1 [hunk 1]
```diff
@@ -55,50 +55,53 @@ function Test-Assessment-25372 {
         Write-PSFMessage "Failed to get GSA device usage summary: $_" -Tag Test -Level Warning
     }
 
-    Write-ZtProgress -Activity $activity -Status 'Getting Intune managed device count'
+    Write-ZtProgress -Activity $activity -Status 'Getting Entra ID managed device count'
 
-    # Query Q2: Count Intune-managed devices
-    $intuneDeviceCount = $null
+    # Query Q2: Count Entra ID managed devices (joined and hybrid joined)
+    $entraDeviceCount = $null
```
[the remark this answers is not in the document]
Comment (ashwinikarke) [reply src/powershell/tests/Test-Assessment.25372.ps1]: Updated.

**Review by alexandair — CHANGES_REQUESTED** [review]
@ashwinikarke 
Could we replace 
```
-RelativeUri "devices?`$filter=trustType eq 'AzureAd' or trustType eq 'ServerAd'&`$count=true&`$top=999&`$select=id,displayName,operatingSystem,operatingSystemVersion,trustType"
```
 with the database query?

Also, what issue # does this PR fix?

**Review by alexandair — CHANGES_REQUESTED** [review]
@ashwinikarke Please, address my feedback.

File: src/powershell/tests/Test-Assessment.25372.ps1 [hunk 2]
```diff
@@ -44,61 +46,73 @@ function Test-Assessment-25372 {
...
-    # Edge case: GSA devices > Intune devices (data inconsistency; GSA has more devices than Intune)
-    if ($totalGsaDevices -gt $totalManagedDevices -and $totalManagedDevices -gt 0) {
+    # Handle any query failure - cannot determine deployment status
+    if ($gsaCmdletFailed -or $deviceQueryFailed) {
+        Write-PSFMessage "Unable to retrieve GSA deployment data due to query failure" -Tag Test -Level Warning
         $customStatus = 'Investigate'
-        $deploymentPercentage = 'N/A'
-        $gap = 'N/A'
-        $testResultMarkdown = "⚠️ Global Secure Access device count exceeds the Intune-managed device count. This indicates stale GSA device records, devices removed from Intune management, or data synchronization issues between systems. Review both data sources to reconcile counts.`n`n%TestResult%"
+        $testResultMarkdown = "⚠️ Unable to determine GSA client deployment coverage due to query failure, connection issues, or insufficient permissions.`n`n"
```
[the remark this answers is not in the document]
Comment (ashwinikarke) [reply src/powershell/tests/Test-Assessment.25372.ps1]: @alexandair Actually, the %TestResult% placeholder was not being replaced because the report generation block was skipped when the queries failed. Therefore, I have removed it.
image

**Review by alexandair — APPROVED** [review]
LGTM
